tests/sdk/flagsmith.test.ts

```diff
@@ -275,18 +276,119 @@ test('getIdentitySegments throws error if identifier is empty string', () => {
 })
 
 
-async function wipeFeatureStateUUIDs (enviromentModel: EnvironmentModel) {
+test('offline_mode', async() => {
+    // Given
+    const environment: EnvironmentModel = environmentModel(JSON.parse(environmentJSON('offline-environment.json')));
+
+    class DummyOfflineHandler extends BaseOfflineHandler {
+        getEnvironment(): EnvironmentModel {
+            return environment;
+        }
+    }
+
+    // When
+    const flagsmith = new Flagsmith({ offlineMode: true, offlineHandler: new DummyOfflineHandler() });
+
+    // Then
+    // we can request the flags from the client successfully
+    const environmentFlags: Flags = await flagsmith.getEnvironmentFlags();
+    let flag = environmentFlags.getFlag('some_feature');
+    expect(flag.isDefault).toBe(false);
+    expect(flag.enabled).toBe(true);
+    expect(flag.value).toBe('offline-value');
+
+
+    const identityFlags: Flags = await flagsmith.getIdentityFlags("identity");
+    flag = identityFlags.getFlag('some_feature');
+    expect(flag.isDefault).toBe(false);
+    expect(flag.enabled).toBe(true);
+    expect(flag.value).toBe('offline-value');
+});
+
+
+test('test_flagsmith_uses_offline_handler_if_set_and_no_api_response', async () => {
+    // Given
+    const environment: EnvironmentModel = environmentModel(JSON.parse(environmentJSON('offline-environment.json')));
+    const api_url = 'http://some.flagsmith.com/api/v1/';
+    const mock_offline_handler = new BaseOfflineHandler() as jest.Mocked<BaseOfflineHandler>;
+  
+    jest.spyOn(mock_offline_handler, 'getEnvironment').mockReturnValue(environment);
+
+    const flagsmith = new Flagsmith({
+      environmentKey: 'some-key',
+      apiUrl: api_url,
+      offlineHandler: mock_offline_handler,
+    });
+
+    jest.spyOn(flagsmith, 'getEnvironmentFlags');
+    jest.spyOn(flagsmith, 'getIdentityFlags');
+
+  
+    flagsmith.environmentFlagsUrl = 'http://some.flagsmith.com/api/v1/environment-flags';
+    flagsmith.identitiesUrl = 'http://some.flagsmith.com/api/v1/identities';
+
+    // Mock a 500 Internal Server Error response
+    const errorResponse = new Response(null, {
+        status: 500,
+        statusText: 'Internal Server Error',
+    });
+
+    // @ts-ignore
+    fetch.mockReturnValue(Promise.resolve(errorResponse));
+
+    // When
+    await flagsmith.getEnvironmentFlags();
+    await flagsmith.getIdentityFlags('identity', {});
+
+    // Then
+    expect(mock_offline_handler.getEnvironment).toHaveBeenCalledTimes(1);
+    expect(flagsmith.getEnvironmentFlags).toHaveBeenCalled();
+    expect(flagsmith.getIdentityFlags).toHaveBeenCalled();
+  
+    const environmentFlags:Flags = await flagsmith.getEnvironmentFlags();
+    const identityFlags:Flags = await flagsmith.getIdentityFlags('identity', {});
```

Comment:
Is there a reason we don't just assign these values in the calls made a few lines above?

Reply:
Oops, I really don't know why. Fixed